integration/benchmark/main.go

```diff
@@ -0,0 +1,234 @@
+		rand.Seed(time.Now().UnixNano())
+		sleepTime := time.Duration(rand.Int63n(50))
+		// fmt.Printf("SleepTime = %d\n", sleepTime)
+		time.Sleep(sleepTime * time.Second / 100)
+	}
+}
+
+// This is a bad test. client.Fetch runtime depends on network speed.
+func BenchmarkGetImage(b *testing.B) {
+	client, err := newClient(b, address)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error new client : %s\n", err)
+		b.Fatal(err)
+	}
+	defer client.Close()
+
+	ctx, cancel := testContext(b)
+	defer cancel()
+
+	_, fetchErr := client.Fetch(ctx, testImage, getRemoteOpts()...)
+	if fetchErr != nil {
+		fmt.Fprintf(os.Stderr, "Error Fetch : %s\n", fetchErr)
+		return
+	}
+
+	_, getErr := client.GetImage(ctx, testImage)
+	if getErr != nil {
+		fmt.Fprintf(os.Stderr, "Error get Image : %s\n", getErr)
+		b.Error(getErr)
+		return
+	}
+}
+
+func BenchmarkContainerCreate(b *testing.B) {
+	client, err := newClient(b, address)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error new client : %s\n", err)
+		b.Fatal(err)
+	}
+	defer client.Close()
+
+	ctx, cancel := testContext(b)
+	defer cancel()
+
+	image, pullErr := client.Pull(ctx, testImage, WithPullUnpack)
+	if pullErr != nil {
+		fmt.Fprintf(os.Stderr, "Error Pull : %s\n", pullErr)
+		b.Error(err)
+		return
+	}
+
+	spec, err := oci.GenerateSpec(ctx, client, &containers.Container{ID: "test"}, oci.WithRootFSPath("/var/lib/containerd-test"), oci.WithImageConfig(image), withTrue())
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error Generate Spec : %s\n", err)
+		b.Error(err)
+		return
+	}
+
+	var containers []Container
+	defer func() {
+		for _, c := range containers {
+			if err := c.Delete(ctx, WithSnapshotCleanup); err != nil {
+				fmt.Fprintf(os.Stderr, "Error Container delete : %s\n", err)
+				b.Error(err)
+			}
+		}
+	}()
+
+	// reset the timer so that only the container creation time is counted
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		id := fmt.Sprintf("%s-%d", "test", i)
+		container, err := client.NewContainer(ctx, id, WithNewSnapshot(id, image), WithSpec(spec))
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Error New Container : %s\n", err)
+			b.Error(err)
+			return
+		}
+		containers = append(containers, container)
+	}
+}
+
+func newClient(t testing.TB, address string, opts ...ClientOpt) (*Client, error) {
+	if testing.Short() {
+		t.Skip()
+	}
+	if rt := os.Getenv("TEST_RUNTIME"); rt != "" {
+		opts = append(opts, WithDefaultRuntime(rt))
+	}
+	// testutil.RequiresRoot(t) is not needed here (already called in TestMain)
+	return New(address, opts...)
+}
+
+func testContext(t testing.TB) (context.Context, context.CancelFunc) {
+	ctx, cancel := context.WithCancel(context.Background())
+	ctx = namespaces.WithNamespace(ctx, testNamespace)
+	if t != nil {
+		ctx = logtest.WithT(ctx, t)
+	}
+	return ctx, cancel
+}
+
+func withTrue() oci.SpecOpts {
+	return oci.WithProcessArgs("true")
+}
+
+func getRemoteOpts() []RemoteOpt {
+	platformList := []string{"linux/amd64", "linux/arm64/v8", "linux/s390x"}
```

Comment:
Why does it have these platforms?

Reply:
@sbuckfelder ?

Reply:
I am removing these platforms pending further information about the necessity of including them.